.travis.yml

```diff
@@ -7,54 +7,76 @@ cache:
 
-# Some of the instructions to install and run minikube are from https://blog.travis-ci.com/2017-10-26-running-kubernetes-on-travis-ci-with-minikube
 before_install:
+  # download things concurrently
+  - |
+    $(
+      curl -Lo dep "https://github.com/golang/dep/releases/download/${DEP_VERSION}/dep-linux-amd64" &
+      curl -Lo install_bazel.sh "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh" &
+      curl -Lo helm.tar.gz "https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}-linux-amd64.tar.gz" &
+      curl -Lo kubectl "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" &
+      wait
+    )
   - |
-    curl -Lo dep "https://github.com/golang/dep/releases/download/${DEP_VERSION}/dep-linux-amd64"
     chmod +x dep
     sudo mv dep /usr/local/bin/
   - |
-    curl -Lo install.sh "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh"
-    chmod +x install.sh
-    ./install.sh --user
-    rm -f install.sh
+    chmod +x install_bazel.sh
+    ./install_bazel.sh --user
+    rm -f install_bazel.sh
     mv .bazelrc.travis .bazelrc
   - |
-    curl -Lo kubectl "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl"
+    tar -zxvf helm.tar.gz
+    sudo mv linux-amd64/helm /usr/local/bin/
+    rm -rf linux-amd64
+  - |
     chmod +x kubectl
     sudo mv kubectl /usr/local/bin/
-  - |
-    go get sigs.k8s.io/kind
-    docker pull kindest/node:${KUBE_VERSION}
-    kind create cluster --image="kindest/node:${KUBE_VERSION}"
 install:
   - make setup-ci
 
 script:
   - make update-bazel
   - make verify
   - make test-ci
+  - go get sigs.k8s.io/kind
   - |
+    kind create cluster --image="kindest/node:${KUBE_VERSION}"
     export KUBECONFIG="$(kind get kubeconfig-path)"
     export KUBECONTEXT="kubernetes-admin@1"
-    make integration-test-ci
+  - |
+    kubectl create clusterrolebinding tiller-cluster-admin \
+      --clusterrole=cluster-admin \
+      --serviceaccount=kube-system:default
+    helm init
+  # run integration tests while tiller is starting
+  - make integration-test-ci
+  - |
+    helm repo add svc-cat https://svc-catalog-charts.storage.googleapis.com
+    helm install svc-cat/catalog --name catalog --namespace catalog
+    helm install svc-cat/ups-broker --name ups-broker --namespace ups-broker
+  # build container and run checks while service catalog and UPS are starting
   - make docker
   - make check
+  - |
+    kubectl create -f it/sc/ups-clusterservicebroker.yaml
+    sleep 5
```

Comment:
Give Service Catalog a few seconds to list classes from broker.

Reply:
add a polling loop like we have for cluster startup in setup_local_cluster